### ISSUE TITLE
Decoration for unselected tabs in TabBar

### ISSUE THREAD

**Issue by hlemcke** [flutter/flutter#75588]
## Use case

Display a tabbar where the tabs look like physical paper tabs with rounded corners top-left and top-right. The selected tab should look "exposed" (e.g. bold text or other color).

## Proposal

Add parameter `unselectedDecoration` to `TabBar`.

TabBar provides `Decoration indicator`. I can use it for example like this:

    TabBar(
      controller: _tabController,
      indicator: BoxDecoration(
        borderRadius: BorderRadius.only(
          topLeft: Radius.circular(10.0),
          topRight: Radius.circular(10.0), ),
        color: Colors.orange,  ),
      tabs: _tabList, );

But there is no option to decorate unselected tabs in a similar way. Trying to decorate the tab itself results in `indicator` not visible, beause it seems to "hide behind the tab".

An alternative would be to give the `TabBar.indicator` a higher precedence than `Tab.indicator` when rendering.

**Comment by MeandNi** [comment]
I am not sure if this will be a problem when the user drags the TabBarView's PageView left or right because the `indicator` is only painted on the top of material tabs.

**Comment by hlemcke** [comment]
Sorry, but the indicator is NOT painted on the top of material tabs. Instead it hides behind the indicator of the tab itself.

**Comment by MeandNi** [comment]
> Sorry, but the indicator is NOT painted on the top of material tabs. Instead it hides behind the indicator of the tab itself. I can provide an example if needed.

thank you for your reply. yeah, I mean the `indicator` is painted by CustomPaint and separated from the tabs. Do you want `unselectedDecoration` to work like this, or just want a way to decorate unselected tabs.

**Comment by hlemcke** [comment]
"unselectedDecoration" does fit better into the architecture of "TabBar" so I prefer that solution.

**Comment by gustavoalves93** [comment]
an unselectedDecoration woud be great for me too... hahaha

**Comment by VovaMedvediev** [comment]
big plus for this feature

**Comment by SakshamKarnawat** [comment]
This is well needed. I wanted to set an indicator/underline to unselected tabs too, but now I'll have to use a Stack (with a lot of setStates) to achieve that.

**Comment by awatson32** [comment]
Would love to be able to add a default color the unselected indicatorColor. Unless there is a more efficient way to do this.

**Comment by Shiba-Kar** [comment]
This would be great

**Comment by ankitparmar007** [comment]
it's still not available, hope see in Flutter 4 !

**Comment by jefferyleo** [comment]
Still no luck for this in 2023

**Comment by benjith-kizhisseri** [comment]
#following

**Comment by MSatyam-Mishra** [comment]
is there any workaround to achieve it???

**Comment by d-khomenko** [comment]
Still no luck for this in 2023

**Comment by HansMuller** [comment]
Marking this as P3: not because it's unimportant or even a bad idea, only because we (still) don't have plans to work on it. A PR that created the feature would be welcome.

**Comment by TusharKhari** [comment]
try this out



class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> with SingleTickerProviderStateMixin {
  late TabController _tabController;
  late PageController _pageController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: 3, vsync: this);
    _pageController = PageController();
    _tabController.addListener(() { 
     if( _tabController.indexIsChanging){
      setState(() {
        
      });
     }
    });
    _tabController.addListener(() {
      _pageController.animateToPage(
        _tabController.index,
        duration: Duration(milliseconds: 300),
        curve: Curves.easeInOut,
      );
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Tab Bar with Indicator'),
        bottom: TabBar(
          controller: _tabController,
          tabs: [
            _CustomTab(
              text: 'Tab 1',
              isSelected: _tabController.index == 0,
            ),
            _CustomTab(
              text: 'Tab 2',
              isSelected: _tabController.index == 1,
            ),
            _CustomTab(
              text: 'Tab 3',
              isSelected: _tabController.index == 2,
            ),
          ],
        ),
      ),
      body: Column(
        children: [
          LinearProgressIndicator(
            value: (_tabController.index + 1) / _tabController.length,
            backgroundColor: Colors.grey[300],
            valueColor: AlwaysStoppedAnimation<Color>(Colors.blue),
          ),
          Expanded(
            child: PageView(
              controller: _pageController,
              onPageChanged: (index) {
                _tabController.animateTo(index);
              },
              children: [
                Container(color: Colors.blue, child: Center(child: Text('Tab 1'))),
                Container(color: Colors.green, child: Center(child: Text('Tab 2'))),
                Container(color: Colors.orange, child: Center(child: Text('Tab 3'))),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _CustomTab extends StatelessWidget {
  final String text;
  final bool isSelected;

  _CustomTab({required this.text, required this.isSelected});

  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(
        color: isSelected ? Colors.red : Colors.white,
        border: Border.all(color: isSelected ? Colors.red : Colors.yellow),
      ),
      child: Center(
        child: Text(
          text,
          style: TextStyle(
            color: isSelected ? Colors.white : Colors.black,
          ),
        ),
      ),
    );
  }
}

**Comment by yeonsh** [comment]
Still no luck for this in 2024

**Comment by azeunkn0wn** [comment]
Yeah. This should do for now. https://github.com/flutter/flutter/issues/75588#issuecomment-1900812085

**Comment by Benjith** [comment]
2025

**Comment by AbdullahChauhan** [comment]
Any update on this?

**Comment by Ayeni-yeniyan** [comment]
I need this but without the custom painter animation. Damn.

**Comment by Ayeni-yeniyan** [comment]
> I hope to hide the unselected label, just like BottomNavigationBar

Hide how? Not sure you can do that with tabs.
